### ISSUE TITLE
Unable to add new column to table after updating Crate version

### ISSUE THREAD

**Issue by tellezb** [crate/crate#6674]
**Environment:**
Crate 2.1.10 (updated from 1.1.6)
Windows 10 Pro 64-bit, Intel
3 nodes all on the same machine

**crate.yml:**
cluster.name: TELLEZ
gateway.expected_nodes: 3
gateway.recover_after_nodes: 2
discovery.zen.minimum_master_nodes: 2
license.enterprise: false
node.max_local_storage_nodes: 3

**Problem description:**
I recently upgraded my local Crate cluster from 1.1.6 to 2.1.10. Migration went well after running OPTIMIZE on some of my tables...data and cluster checks were all green and database functioned as usual.

I then attempted to add a column to one of my tables: `ALTER TABLE ALERT_DEVICE_EVENTS ADD ALERT_GROUP_ID LONG`. I received the following exception: `SQLParseException: Can't parse [index] value [false] for field [alert_partition], expected [true] or [false]`, but the new column *appears* to have been added to the table. (NOTE: alert_partition is the partition column for this table)

Now, attempting to UPDATE the new ALERT_GROUP_ID column in this table throws exception: `StrictDynamicMappingException: mapping set to strict, dynamic introduction of [alert_group_id] within [default] is not allowed`.

What is happening?

**Comment by seut** [comment]
@tellezb as I cannot reproduce your issue, can you please paste us your table schema? or send it to me sebastian [at] crate.io.
also on which crate version was this table created on?
did you upgrade to 2.1.10 without any further optimization run first?

**Comment by tellezb** [comment]
@seut I emailed you the full table schema. It was originally created in Crate version 1.1.6. Our production cluster created this table in version 1.0.5.

I updated the cluster to 2.1.10. After upgrading binaries:
- I added node.attr.max_local_storage_nodes to the config yaml
- I added license.enterprise: false to the config yaml
- I created log4j2.properties
- I rearranged the data directory by removing the cluster name from the path
- I ran `OPTIMIZE TABLE alert_device_events WITH (upgrade_segments=true);` (as well as for a few other tables)
